### ISSUE TITLE
Date picker is not closing

### ISSUE THREAD

**Issue by aureosouza** [henninghall/react-native-date-picker#601]
**Describe the bug**
Modal date picker is not closing when `open` prop changes to `false`

**Expected behavior**
Modal date picker should be closed

**To Reproduce**
Demo link with issue: https://github.com/aureosouza/demo-date-picker
```javascript
  const [date, setDate] = useState(new Date());
  const [open, setOpen] = useState(false);

  useEffect(() => {
    if (open) {
      setTimeout(() => {
        setOpen(false); // should close automatically in 3 sec, but it currently doesn't
      }, 3000);
    }
  }, [open]);

  return (
    <SafeAreaView>
      <Button title="Open" onPress={() => setOpen(true)} />
      <DatePicker
        modal
        open={open}
        date={date}
        onConfirm={(date) => {
          setOpen(false);
          setDate(date);
        }}
        onCancel={() => {
          setOpen(false);
        }}
      />
    </SafeAreaView>
  );
```

**Smartphone:**
 - OS: iOS
 - React Native version: 0.70.5
 - react-native-date-picker version: 4.2.6

**Comment by aureosouza** [comment]
@henninghall opened a [PR](https://github.com/henninghall/react-native-date-picker/pull/602), thanks!

**Comment by henninghall** [comment]
Thanks for your contribution! released the fix in version `4.2.7`
